### PR TITLE
Add example to use libRmlUi with SDL2 SDL_Renderer / SDL_RenderGeometry

### PR DESCRIPTION
**Pull request by 1bsyl**
for request  #250 
@baidwwy
I've updated the sample done for libRocket to match RmlUI sample/basics/sdl2 
quickly tested.

### CODE REVIEW

**Comment by baidwwy** [comment]
That’s great. Thanks

**Review by mikke89 — COMMENTED** [review]
Very interesting, thank you!

Not sure how to handle this with regard to the CMake list. I guess it can be a bit confusing wrt. to the old SDL2 sample. Maybe rename the old one to eg. `SDL2OpenGL`, and add this as a separate sample? And further we should only enable this new one after testing the SDL version, after this has had a release.

Let me know if any of you have thoughts in this regard. For now it should be perfectly fine to merge it as is (after fixing the minor suggestions), and we can work on the CMake and sample things later, cheers!

File: Samples/basic/sdl2_sdlrenderer/src/RenderInterfaceSDL2.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ */
+
+#include <RmlUi/Core.h>
+#include <SDL_image.h>
+
+#include "RenderInterfaceSDL2.h"
+
+#if !(SDL_VIDEO_RENDER_OGL)
+    #error "Only the opengl sdl backend is supported."
+#endif
```
[the remark this answers is not in the document]
Comment (1bsyl) [reply Samples/basic/sdl2_sdlrenderer/src/RenderInterfaceSDL2.cpp]: yep done

File: Samples/basic/sdl2_sdlrenderer/src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+{
+#ifdef RMLUI_PLATFORM_WIN32
+	AllocConsole();
+#endif
+
+    int window_width = 1024;
+    int window_height = 768;
+
+    SDL_Init( SDL_INIT_VIDEO );
+    SDL_Window * screen = SDL_CreateWindow("LibRocket SDL2 with SDL_Renderer test", 20, 20, window_width, window_height, SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE);
```
[the remark this answers is not in the document]
Comment (1bsyl) [reply Samples/basic/sdl2_sdlrenderer/src/main.cpp]: yep done

**Comment by 1bsyl** [comment]
I've just update the PR.
SDL2 is expected to be released soon, so you'll be able to try it.
Then fell free to change whatever you want.

**Comment by mikke89** [comment]
Perfect, thanks! I'll hold on with merging until I've had a chance to test with the release of the next version of SDL2.

**Comment by aquawicket** [comment]
I have the SDL2 File Interfaces for this, then we can leave out the shell dependency make it strictly SDL2+RmlUi.

**Comment by aquawicket** [comment]
Firstly, thank you 1bsyl for providing this path and your help over at SDL.  I went ahead dove into this and it works. So far, I've made a few fixes. Changed from using viewports to raw position drawing. Removed OpenGL hints so D3D, Metal and Vulkan can take control. Pulled in the Shell File Interface as part of the base code. Images are working well.  Also added gif animation support. I've run it on successfully on Windows 10 x86 / x86_64(opengl/d3d/Vulcan), Mac 11.5.2 x86_64(Metal/opengl),  Lxle Linux i386(opengl), Ubuntu x86_64(opengl), Raspberry Pi 4 armhf (opengles), and i'm currently working on iOS(metal/opengles) and Android(opengles). I created a separate repository to keep down the clutter here until we have a viable PR. 
There is no CMake or build scripts as of yet, but it uses the same as the RmlUI/Samples/basic/sdl2 project, minus SDL2 addons (SDL_Image, SDL_tft, etc). don't mind the practice samples. Still early code, but looking very promising.  [SDL2RendererRml](https://github.com/aquawicket/SDL2RendererRml)

**Comment by mikke89** [comment]
@1bsyl Thanks again! This works for me, I'm also in the process of adding CMake integration.

@aquawicket Those enhancements sounds very interesting! I'm very much looking forward to the PR(s) ;)

**Comment by mikke89** [comment]
For reference, see also SDL2 PR https://github.com/libsdl-org/SDL/pull/4195.

**Pull request merged**